cmd/cosign/cli/verify_manifest.go

```diff
@@ -116,95 +110,91 @@ func (c *VerifyManifestCommand) Exec(ctx context.Context, args []string) error {
+							Image string
+						}
+					}
+				}
+			}
 		}
+	}
+}
 
-		decoded, _, err := deserializer.Decode(ext.Raw, nil, nil)
-		if err != nil {
-			return images, errors.New("unable to decode the manifest")
+func (ic *imageContainer) images() []string {
+	images := []string(nil)
+	var addImage = func(image string) {
+		if len(image) > 0 {
+			images = append(images, image)
 		}
+	}
 
-		var (
-			d   *appsv1.Deployment
-			rs  *appsv1.ReplicaSet
-			ss  *appsv1.StatefulSet
-			ds  *appsv1.DaemonSet
-			job *v1beta1.CronJob
-			pod *corev1.Pod
-		)
-		containers := make([]corev1.Container, 0)
-		switch obj := decoded.(type) {
-		case *appsv1.Deployment:
-			d = obj
-			containers = append(containers, d.Spec.Template.Spec.Containers...)
-			containers = append(containers, d.Spec.Template.Spec.InitContainers...)
-			for _, c := range containers {
-				images = append(images, c.Image)
-			}
-		case *appsv1.DaemonSet:
-			ds = obj
-			containers = append(containers, ds.Spec.Template.Spec.Containers...)
-			containers = append(containers, ds.Spec.Template.Spec.InitContainers...)
-			for _, c := range containers {
-				images = append(images, c.Image)
-			}
-		case *appsv1.ReplicaSet:
-			rs = obj
-			containers = append(containers, rs.Spec.Template.Spec.Containers...)
-			containers = append(containers, rs.Spec.Template.Spec.InitContainers...)
-			for _, c := range containers {
-				images = append(images, c.Image)
-			}
-		case *appsv1.StatefulSet:
-			ss = obj
-			containers = append(containers, ss.Spec.Template.Spec.Containers...)
-			containers = append(containers, ss.Spec.Template.Spec.InitContainers...)
-			for _, c := range containers {
-				images = append(images, c.Image)
-			}
+	// Pod
+	for _, c := range ic.Spec.InitContainers {
+		addImage(c.Image)
+	}
+	for _, c := range ic.Spec.Containers {
+		addImage(c.Image)
+	}
 
-		case *v1beta1.CronJob:
-			job = obj
-			containers = append(containers, job.Spec.JobTemplate.Spec.Template.Spec.Containers...)
-			containers = append(containers, job.Spec.JobTemplate.Spec.Template.Spec.InitContainers...)
-			for _, c := range containers {
-				images = append(images, c.Image)
-			}
-		case *corev1.Pod:
-			pod = obj
-			containers = append(containers, pod.Spec.Containers...)
-			containers = append(containers, pod.Spec.InitContainers...)
+	// Deployment, ReplicaSet, StatefulSet, DaemonSet, Job
+	for _, c := range ic.Spec.Template.Spec.InitContainers {
+		addImage(c.Image)
+	}
+	for _, c := range ic.Spec.Template.Spec.Containers {
+		addImage(c.Image)
+	}
+
+	// CronJob
+	for _, c := range ic.Spec.JobTemplate.Spec.Template.Spec.InitContainers {
+		addImage(c.Image)
+	}
+	for _, c := range ic.Spec.JobTemplate.Spec.Template.Spec.Containers {
+		addImage(c.Image)
+	}
 
-			for _, c := range containers {
-				images = append(images, c.Image)
+	return images
+}
+
+func getImagesFromYamlManifest(manifest []byte) ([]string, error) {
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(manifest), 4096)
```

Comment:
I'm sort of puzzled by how this works without the annotations 🤔

Reply:
the marshaler assumes `camelCase` for the keys, the json keys are only needed when you want to control a behavior or  key name (like you want `init_containers` vs the default `initContainers`).